skcosmo/decomposition/_kernel_pcovr.py

```diff
@@ -200,31 +214,54 @@ def __init__(
 
         self.mixing = mixing
         self.n_components = n_components
-        self.alpha = alpha
 
         self.svd_solver = svd_solver
         self.tol = tol
         self.iterated_power = iterated_power
         self.random_state = random_state
         self.center = center
 
-        self.kernel = kernel
-        self.kernel_params = kernel_params
-        self.gamma = gamma
-        self.degree = degree
-        self.coef0 = coef0
+        if regressor is None:
+            regressor = KernelRidge(
+                alpha=1e-6,
+                kernel="linear",
+                gamma=None,
+                degree=3,
+                coef0=1,
+                kernel_params=None,
+            )
```

Comment:
Per sklearn guidelines, variable checking should only be in the `fit` call and never in `__init__`. What I would suggest is either setting the default regressor to what you have here or setting it in the `fit` call. Alternatively, it would be within the behavior of sklearn to have `default = None` and then during `fit` set the regressor using the KPCovR kernel parameters. I know you've had a whole discussion on this, but what you have here isn't in line with sklearn guidelines.